src/peft/tuners/xlora/model.py

```diff
@@ -0,0 +1,345 @@
+        """
+        Get the current top_k LoRA experts value.
+        """
+        classifier: xLoRAClassifier = self.internal_xlora_classifier  # type: ignore
+        return classifier.config.top_k_lora
+
+    def set_global_scaling_weight(self, weight: float):
+        """
+        Set the global LoRA weight, a scalar to multiply the output of each LoRA adapter by. This is by default 1. This is reflected in the config.
+        """
+        classifier: xLoRAClassifier = self.internal_xlora_classifier  # type: ignore
+        classifier.config.global_scaling_weight = weight
+
+    def get_global_scaling_weight(self) -> float:
+        """
+        Get the global LoRA weight.
+        """
+        classifier: xLoRAClassifier = self.internal_xlora_classifier  # type: ignore
+        return classifier.config.global_scaling_weight
+
+    def get_latest_scalings(self) -> Optional[torch.Tensor]:
+        """
+        Returns the latest scalings prediction, or None if no scalings have been predicted. The tensor is of shape (batch_size, seq_len, n_layers, n_classes).
+        """
+        return self.internal_xlora_scalings
+
+    def get_scalings_log(self) -> List[torch.Tensor]:
+        """
+        Returns a shallow (only copying the list itself not the tensors) copy of the list containing the scalings log. Editing the list does not change the underlying log.
+        The tensors are of shape (batch_size, seq_len, n_layers, n_classes). The seq_len dim may vary with input dimension.
+        """
+        classifier: xLoRAClassifier = self.internal_xlora_classifier  # type: ignore
+        return classifier.log_scalings.copy()
+
+    def set_scaling_pass_value(self, value: Union[Number, None]):
+        """
+        Manually set the scalings to a specific value during the scaling pass, forever. Call this function with None to enable the default
+        scalings.
+
+        This is reflected in the config.
+        """
+        classifier: xLoRAClassifier = self.internal_xlora_classifier  # type: ignore
+        classifier.set_override_scaling_pass_value(value)
+
+    def print_scalings_predictions(self, n_predictions_lifetime: int):
+        """
+        Print the scaling states for the next n classifier predictions (i.e. forward, generate passes)
+        """
+        classifier: xLoRAClassifier = self.internal_xlora_classifier  # type: ignore
+        classifier.n_predictions_lifetime = n_predictions_lifetime
+
+    def enable_scalings_logging(self):
+        """
+        Enable scalings logging.
+        """
+        classifier: xLoRAClassifier = self.internal_xlora_classifier  # type: ignore
+        classifier.scalings_logging = True
+
+    def disable_scalings_logging(self):
+        """
+        Disable scalings logging, clearing the log.
+        """
+        classifier: xLoRAClassifier = self.internal_xlora_classifier  # type: ignore
+        classifier.scalings_logging = False
+        classifier.log_scalings = []
```

Comment:
How about `self.log_scalings.clear()` in case someone else holds a reference to it?